### DISCUSSION TITLE
How to set cookie in response?

### DISCUSSION THREAD

**Discussion by goldwind-ting** [tokio-rs/axum#351] [Q&A] [answered]

**Comment by davidpdrsn** [comment, marked as answer]
There are many ways to add headers to a response. See [here](https://docs.rs/axum/0.2.3/axum/index.html#building-responses) for examples.

The easiest way is to use `axum::response::Headers`:

```rust
use axum::{
    http::header::SET_COOKIE,
    response::{Headers, Html, IntoResponse},
};

async fn handler() -> impl IntoResponse {
    let headers = Headers([(SET_COOKIE, "key=value")]);
    let content = Html("<h1>Hello, World!</h1>");
    (headers, content)
}
```

**Reply by goldwind-ting** [reply]
Thank you for this!!

**Reply by k7575** [reply]
if many cookies?

**Reply by davidpdrsn** [reply]
@k7575 https://docs.rs/axum-extra/latest/axum_extra/extract/cookie/index.html

**Reply by merlinaudio** [reply]
FYI for googlers: this is now called `AppendHeaders`
